src/bft.py

```diff
@@ -12,4 +12,16 @@ def bf_order(t: T | None) -> Iterable[int]:
     >>> list(bf_order(tree))
     [2, 1, 4, 3, 5]
     """
-    return  # FIXME
+    queue = deque()
+    output = []
+    if not t:
+        return []
```

Comment:
I would flip the order here. If `t is None`, then there is no need to build the `queue`. It is an annoying special case, but hard to avoid. YOu can, though, also drop the test for `t is None` and return and simply not put `t` in the `queue` if `t` is `None`. That way, the empty `output` list you create will be the (correct) result.

```python
output, queue = [], deque([t] if t else [])
while queue:
    ...
return output
```